src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/ReleasePrisonerService.kt

```diff
@@ -0,0 +1,61 @@
+package uk.gov.justice.hmpps.prison.service.enteringandleaving
+
+import org.springframework.data.repository.findByIdOrNull
+import uk.gov.justice.hmpps.prison.api.model.InmateDetail
+import uk.gov.justice.hmpps.prison.api.model.RequestToReleasePrisoner
+import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyLocation
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderBooking
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.AgencyLocationRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBookingRepository
+import uk.gov.justice.hmpps.prison.service.BadRequestException
+import uk.gov.justice.hmpps.prison.service.EntityNotFoundException
+import uk.gov.justice.hmpps.prison.service.transformers.OffenderTransformer
+import kotlin.Result.Companion.failure
+import kotlin.Result.Companion.success
+
+class ReleasePrisonerService(
```

Comment:
you've wired up a repository here - should the service therefore be transactional at all?